src/components/Trade/hooks/useSwapper/useSwapper.ts

```diff
@@ -257,68 +277,127 @@ export const useSwapper = () => {
   }
 
   const updateQuoteDebounced = useRef(
-    debounce(async ({ amount, sellAsset, buyAsset, action, wallet, swapperManager }) => {
-      try {
-        const swapper = await swapperManager.getBestSwapper({
-          buyAssetId: buyAsset.assetId,
-          sellAssetId: sellAsset.assetId,
-        })
+    debounce(
+      async ({
+        amount,
+        sellAsset,
+        buyAsset,
+        action,
+        wallet,
+        swapperManager,
+        accountSpecifiersList,
+        chainAdapterManager,
+      }: DebouncedQuoteInput) => {
+        try {
+          const swapper = await swapperManager.getBestSwapper({
+            buyAssetId: buyAsset.assetId,
+            sellAssetId: sellAsset.assetId,
+          })
 
-        if (!swapper) throw new Error('no swapper available')
+          if (!swapper) throw new Error('no swapper available')
+          const [sellAssetUsdRate, buyAssetUsdRate, feeAssetUsdRate] = await Promise.all([
+            swapper.getUsdRate({ ...sellAsset }),
+            swapper.getUsdRate({ ...buyAsset }),
+            swapper.getUsdRate({ ...feeAsset }),
+          ])
+
+          const { sellAmount, buyAmount, fiatSellAmount } = await calculateAmounts({
+            amount,
+            buyAsset,
+            sellAsset,
+            buyAssetUsdRate,
+            sellAssetUsdRate,
+            action,
+          })
 
-        const [sellAssetUsdRate, buyAssetUsdRate, feeAssetUsdRate] = await Promise.all([
-          swapper.getUsdRate({ ...sellAsset }),
-          swapper.getUsdRate({ ...buyAsset }),
-          swapper.getUsdRate({ ...feeAsset }),
-        ])
+          const { chainId: receiveAddressChainId } = fromAssetId(buyAsset.assetId)
 
-        const { sellAmount, buyAmount, fiatSellAmount } = await calculateAmounts({
-          amount,
-          buyAsset,
-          sellAsset,
-          buyAssetUsdRate,
-          sellAssetUsdRate,
-          action,
-        })
+          const chainAdapter = chainAdapterManager.get(receiveAddressChainId)
```

Comment:
```suggestion
          const chainAdapter = getChainAdapters().get(receiveAddressChainId)
```

to avoid plumbing things around